### ISSUE TITLE
GameElement special values 0 vs -1

### ISSUE THREAD

**Issue by wsu-cb** [CyberBoardPBEM/cbwindows#53]
I see that `CGamDoc::GetGameElementCodeForObject()` returns `(GameElement)0` for inappropriate inputs, but 'CGamDoc::GetVerifiedGameElementCodeForObject()' returns `(GameElement)-1`.  Why the difference?

For what it's worth, it looks to me like `(GameElement)0` would be the result of `MakePieceElement(PieceID(0))`, which suggests to me that `(GameElement)0` isn't a good "error" value because it's actually valid in some cases.  On the other hand, MarkID(0xFFFF) should never be valid, so `(GameElement)-1` looks like an appropriate "error" value.

In short, could we replace uses of `(GameElement)0` with `(GameElement)1`?

**Comment by DLLarson** [comment]
>In short, could we replace uses of (GameElement)0 with (GameElement)1?

Yes... This was an error.

In fact it seems like an assert() should be added since I don't think `GetGameElementCodeForObject` should ever take the default failure path unless the data is corrupted. 

But since it's a code possibility it should be `(GameElement)-1` as you suggest. The callers don't appear check the return value. 

-Dale
